test/test_integration_management.cxx

```diff
@@ -884,6 +884,66 @@ TEST_CASE("integration: user management", "[integration]")
         REQUIRE_SUCCESS(resp.ctx.ec);
         REQUIRE(resp.roles.size() > 0);
     }
+
+    SECTION("change user password")
+    {
+        asio::io_context io;
+        auto guard = asio::make_work_guard(io);
+        std::thread io_thread([&io]() { io.run(); });
+        auto user_name = test::utils::uniq_id("newUser");
+
+        // create options
+        auto options_original = couchbase::cluster_options(integration.ctx.username, integration.ctx.password);
+        auto options_outdated = couchbase::cluster_options(user_name, integration.ctx.password);
+        auto options_updated = couchbase::cluster_options(user_name, "newPassword");
+
+        // create new user and upsert
+        couchbase::core::management::rbac::user new_user{ user_name };
+        new_user.display_name = "change_password_user";
+        new_user.password = integration.ctx.password;
+        new_user.roles = {
+            couchbase::core::management::rbac::role{ "admin" },
+        };
+        auto [cluster, ec] = couchbase::cluster::connect(io, integration.ctx.connection_string, options_original).get();
+        auto coreCluster = cluster.core();
+        couchbase::core::operations::management::user_upsert_request upsertReq{};
+        upsertReq.user = new_user;
+        auto upsertResp = test::utils::execute(coreCluster, upsertReq);
+        REQUIRE_SUCCESS(upsertResp.ctx.ec);
+        usleep(100000); // Sleep so server can process
+        cluster.close();
+        guard.reset();
+        io_thread.join();
+
+        // Make connection with new creds and change password:
+        asio::io_context io2;
+        auto guard2 = asio::make_work_guard(io2);
+        std::thread io_thread2([&io2]() { io2.run(); });
+        auto [cluster_new, ec_new] = couchbase::cluster::connect(io2, integration.ctx.connection_string, options_outdated).get();
+        auto coreCluster_new = cluster_new.core();
+        couchbase::core::operations::management::change_password_request changePasswordReq{};
+        changePasswordReq.newPassword = "newPassword";
+        auto changePasswordResp = test::utils::execute(coreCluster_new, changePasswordReq);
+        REQUIRE_SUCCESS(changePasswordResp.ctx.ec);
+        usleep(100000);
```

Comment:
Same comment on the `usleep`.   Here, though, I don't suppose we can ping the server until the change happens -- I can't imagine we send the password back.   It's unfortunate there isn't a `mutation_token` equivalent that these sort of management calls would return, to allow us a deterministic way of knowing...

Reply:
In the same vein as waiting for the user upsert we could "_wait until cluster connects_" by attempting (and retrying) a new connection with the cluster using the new credentials? Or even do that inside the test:
_while_ (cluster doesn't connect using new password) _do_ (initiate connection with new password) with timeout.

Seems a bit heavy for this use but it would be less arbitrary than sleep, what do you think?

Reply:
Yea -- I do like that approach - good idea.